applications/openshift/general/oauth_login_template_set/policy/stig/shared.yml

```diff
@@ -0,0 +1,60 @@
+
+  If the DOD notice and consent banner is not displayed, this is a finding.
+
+fixtext: |-
+  The following steps may be used to provide a custom login page that will
+  display the DOD notice and consent banner. This will replace OpenShift's
+  default login page. Please note that depending on the number and type
+  of identity providers configured, it is possible that these pages will
+  not be used. If that is the case, then it is the responsibility of the
+  identity provider being used to display the DOD notice and consent banner.
+
+  1. Create a login page template
+
+  > oc adm create-login-template > login.html
+
+  2. Edit the template to add the DOD notice and consent banner by adding
+  the following lines at the end of the html body. In addition, any look
+  and feel custimizations should also be applied to the pages.
+
+      <script type="text/javascript">
+        window.onload = function(){
+                  alert("You are accessing a U.S. Government (USG)
+                  Information System (IS) that is provided for USG-authorized
+                  use only. By using this IS (which includes any device
+                  attached to this IS), you consent to the following
+                  conditions:\n\n-The USG routinely intercepts and monitors
+                  communications on this IS for purposes including, but
+                  not limited to, penetration testing, COMSEC monitoring,
+                  network operations and defense, personnel misconduct
+                  (PM), law enforcement (LE), and counterintelligence (CI)
+                  investigations.\n\n-At any time, the USG may inspect and
+                  seize data stored on this IS.\n\n-Communications using,
+                  or data stored on, this IS are not private, are subject to
+                  routine monitoring, interception, and search, and may be
+                  disclosed or used for any USG-authorized purpose.\n\n-This
+                  IS includes security measures (e.g., authentication and
+                  access controls) to protect USG interests--not for your
+                  personal benefit or privacy.\n\n-Notwithstanding the above,
+                  using this IS does not constitute consent to PM, LE or
+                  CI investigative searching or monitoring of the content
+                  of privileged communications, or work product, related
+                  to personal representation or services by attorneys,
+                  psychotherapists, or clergy, and their assistants. Such
+                  communications and work product are private and
+                  confidential. See User Agreement for details.");
+        }
+      </script>
+
+  3. Add custom login pages to OpenShift
+
+  > oc create secret generic login-template --from-file=login.html -n openshift-config
+  > oc patch oauths cluster --type='merge' -p '{"spec":{"templates":{"login":{"name":"login-template"}}}}'
```

Comment:
SRG-APP-000069-CTR-000125 has a 4th step that's missing here?

Reply:
You mean in the SRG draft (btw the latest draft is [here](https://docs.google.com/spreadsheets/d/1ouRalvXzRHuhPPN_AKMvqqscUqGvVOkkEdMP3YYBO44/edit#gid=1859516253)). ?

That's because SRG-APP-000069-CTR-000125 has the same steps as 120, but the draft conflates all rules into a single check and fix (which it shouldn't do I think, it was just that whoever wrote that draft didn't realized that it would be different checks in cac..), so there is an extra step because both login and provider template are fetched.

I thought that instead of repeating the steps in the `fixtext` it should be enough to point to 120 in the fix. But I don't really feel strongly about this, I can just copy-paste the check and fix from the sheet into 125.

Reply:
btw for some reason one of the policy files I wrote went 1-2-4 and the other had steps 1-2-3. I fixed this.